src/Hooks/VaporWorkCommandFinishedListener.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace Laravel\Nightwatch\Hooks;
+
+use Illuminate\Console\Events\CommandFinished;
+use Laravel\Nightwatch\Core;
+use Laravel\Nightwatch\State\CommandState;
+use Throwable;
+
+/**
+ * @internal
+ */
+final class VaporWorkCommandFinishedListener
+{
+    /**
+     * @param  Core<CommandState>  $nightwatch
+     */
+    public function __construct(
+        private Core $nightwatch,
+    ) {
+        //
+    }
+
+    public function __invoke(CommandFinished $event): void
+    {
+        try {
+            if ($event->command === 'vapor:work') {
+                $this->nightwatch->digest();
+            }
+        } catch (Throwable $e) {
+            $this->nightwatch->report($e, handled: true);
+        }
+    }
```

Comment:
We also shouldn't need the try / catch here.

The `digest` method already handles exceptions, so is safe to just call anywhere (confusing, I know).